### PR TITLE
Add ignorePATH option

### PR DESCRIPTION
**Pull request by soiamsoNG**
Add option to ignore haskell-language-server on PATH
relate to #461 #500

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -187,8 +187,15 @@ async function activateServerForFolder(context: ExtensionContext, uri: Uri, fold
 
   let serverExecutable;
   try {
+    let localServer: string | null;
+    if (workspace.getConfiguration('haskell').get('ignorePATH') === true) {
+      localServer = null;
+      logger.info('Ignoring haskell-language-server on PATH');
+    } else {
+      localServer = findLocalServer(context, logger, uri, folder);
+    }
```
Comment (fendor) [inline src/extension.ts:196]: Shouldn't this logic happen in `findLocalServer`?

Maybe pass in an extra option?

File: package.json [hunk 2]
```diff
@@ -429,4 +435,4 @@
     "which": "^2.0.1",
     "yauzl": "^2.10.0"
   }
-}
+}
```
Comment (fendor) [inline package.json:438]: Unintended change?

**Review by fendor — CHANGES_REQUESTED** [review]
Thank you for this feature! some minor comments open for discussion :)

File: package.json (hunk 2 above)
Comment (soiamsoNG) [reply package.json:438]: vscode 1.64.2 json  formatter trim the tail \n

Comment (fendor) [reply package.json:438]: Colour me surprised

Comment (fendor) [reply package.json:438]: In that case, thank you for correcting it :)

File: src/extension.ts (hunk 1 above)
Comment (soiamsoNG) [reply src/extension.ts]: Thanks for your advice, revised by [8c02da9](https://github.com/haskell/vscode-haskell/pull/534/commits/8c02da90226ca7ae4d8cc6a809a2747c17dfed2e)

**Review by fendor — APPROVED** [review]
LGTM, thank you!

**Comment by jneira** [comment]
feel free to merge when you consider

**Pull request merged**